### DISCUSSION TITLE
What is the best approach for Product Entity in an e-Commerce system?

### DISCUSSION THREAD

**Discussion by ali-gol** [abpframework/abp#10600] [Q&A]
Hi,

I have an e-Commerce platform project and i want to use the abp for my project. I am just thinking about designing for now, haven't started to developer the application. Let me tell you my question:

There is a Product entity which has the Supplier(actually they should be Tenant i believe). So when a tenant login to the page he should see only his products for deleting-modifying his existing products. That's what MultiTenancy feature is doing already.

But when an end-user logins or visits the application without authentication; all end-users should be able to see the whole products. Just like typical of e-commerce systems. In e-bay you have suppliers or tenants. But there is also us(end users, buyers). And we see all the products, but Tenants can't see each others data. I mean even though Product has an entity, a regular user should access to all the products. But if a tenant wants to manage the products he should just his products.

### **What is the best approach here?**
- Can we change default tenant behavior? I want to filter it automatically only when current user role is Supplier or Tenant.
- Or can we solve this problem by using Organizational Units? 
- Or should I use the IMultiTenant but disable to auto filtering feature and apply this logic in my ApplicationServices by applying custom filters to the Repository?

### **Another Question:**
According to documentation: https://aspnetboilerplate.com/Pages/Documents/Multi-Tenancy#host-vs-tenant there is a HostUser and TenantUser. So, if user is logged in with no TenantId ABP assumes it's a host user. And i think HostUsers are allowed to manage to tenants and other staffs. But I also have other type of user which are considered as Customers. So what is best approach here? Or am I lack of some information? Could you please guide me here?

> I just want you to think about a typical eCommerce system which has the suppliers(Tenants) and their products, end-users, HostUsers(BackOffice), and AnonymousUsers which haven't logged in it all, just seeing the products in homepage.

Thank you

**Comment by ismcagdas** [comment]
I think your product shouldn't be a multi tenant entity. Instead, you can add an SupplierId (or OwnerTenantId) field to your Product entity and manually filter records on Product listing page for Suppliers. In that way, buyers can see all Products in the main app.

**Comment by sashok2k** [comment]
> I think your product shouldn't be a multi tenant entity. Instead, you can add an SupplierId (or OwnerTenantId) field to your Product entity and manually filter records on Product listing page for Suppliers. In that way, buyers can see all Products in the main app. 

On my opinion multiple tenants solution is correct decision. 

To disable filtering just use IDataFilter concept.

https://docs.abp.io/en/abp/latest/Data-Filtering
